common/src/chain/transaction/input.rs

```diff
@@ -1,22 +1,40 @@
-use crate::chain::transaction::Transaction;
+use crate::chain::{block::Block, transaction::Transaction};
 use crate::primitives::Id;
 use parity_scale_codec::{Decode, Encode};
 
+#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
+pub enum OutPointSourceId {
+    RegularTransaction(Id<Transaction>),
```

Comment:
What other kinds of transactions are we going to have?

Reply:
Only this. But this is called regular as opposed to block reward, if this is what you're asking. I don't have a preference though.

Reply:
In my mind the word regular implied that we'd have irregular or special transactions in addition to regular transactions but in all honesty I don't care what's it called.